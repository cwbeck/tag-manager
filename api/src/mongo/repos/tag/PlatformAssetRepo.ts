import { injectable } from 'inversify';
import PlatformAsset from '../../models/tag/PlatformAsset';
import UnderPlatformRevisionControl from './abstractions/UnderPlatformRevisionControl';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class PlatformAssetRepo extends UnderPlatformRevisionControl<PlatformAsset> {
    protected readonly auditEnabled = true;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                ___persisting_id: 1,
                _revision_id: 1,
            },
            unique: true,
        },
    ];
}
