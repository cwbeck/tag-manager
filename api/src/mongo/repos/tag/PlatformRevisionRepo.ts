import Repo from '../../abstractions/Repo';
import { injectable } from 'inversify';
import PlatformRevision from '../../models/tag/PlatformRevision';

@injectable()
export default class PlatformRevisionRepo extends Repo<PlatformRevision> {
    protected readonly auditEnabled = true;
}
