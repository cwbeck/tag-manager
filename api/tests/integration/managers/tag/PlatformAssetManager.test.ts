import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import PlatformRevisionRepo from '../../../../src/mongo/repos/tag/PlatformRevisionRepo';
import PlatformAssetRepo from '../../../../src/mongo/repos/tag/PlatformAssetRepo';
import Mock = jest.Mock;

describe('PlatformAssetManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches a platform asset via platform', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();
        ManagerTestUtils.mockOrgUserWithViewAccessOnPlatformRevision();

        const mockPlatformRevision = TestUtils.buildMockPlatformRevision();
        const mockPlatformRevisionRepo = {} as PlatformRevisionRepo;
        mockPlatformRevisionRepo.findByIdThrows = jest.fn().mockResolvedValue(mockPlatformRevision);
        const mockSutModel = TestUtils.buildMockPlatformAsset();
        const mockSutRepo = {} as PlatformAssetRepo;
        mockSutRepo.find = jest.fn().mockResolvedValue([mockSutModel]);
        mockSutRepo.findOneThrows = jest.fn().mockResolvedValue(mockSutModel);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockImplementation((model: any) => {
            if (model.name === 'PlatformAsset') {
                return mockSutRepo;
            }
            return mockPlatformRevisionRepo;
        });

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getPlatformRevision(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    platform_assets {
                        id
                        name
                    }
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
