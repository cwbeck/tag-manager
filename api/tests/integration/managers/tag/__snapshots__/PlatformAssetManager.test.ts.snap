// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformAssetManager fetches a platform asset via platform 1`] = `
Object {
  "data": Object {
    "getPlatformRevision": Object {
      "id": "5edf4c48e9615d9a1c66cbbb",
      "platform_assets": Array [
        Object {
          "id": "5edf4c48e9615d9a1c660ccc",
          "name": "test platform asset",
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
